src/mongoose_rabbit_worker.erl

```diff
@@ -0,0 +1,302 @@
+    MaxMsgQueueLen = proplists:get_value(max_queue_len, Opts),
+    {noreply, State#state{host = Host, amqp_client_opts = AMQPClientOpts,
+                          connection = Connection, channel = Channel,
+                          confirms = IsConfirmEnabled,
+                          max_queue_len = MaxMsgQueueLen}}.
+
+-spec handle_amqp_publish(Method :: mongoose_amqp:method(),
+                          Payload :: mongoose_amqp:message(),
+                          Opts :: worker_opts()) -> {noreply, worker_opts()}.
+handle_amqp_publish(Method, Payload, Opts = #state{host = Host}) ->
+    {PublishTime, Result} =
+        timer:tc(fun publish_message_and_wait_for_confirm/3,
+                 [Method, Payload, Opts]),
+    case Result of
+        true ->
+            update_messages_published_metrics(Host, PublishTime, Payload),
+            ?DEBUG("event=rabbit_message_sent message=~1000p", [Payload]),
+            {noreply, Opts};
+        false ->
+            update_messages_failed_metrics(Host),
+            ?WARNING_MSG("event=rabbit_message_sent_failed reason=negative_ack",
+                         []),
+            {noreply, Opts};
+        {channel_exception, Error, Reason} ->
+            update_messages_failed_metrics(Host),
+            ?WARNING_MSG("event=rabbit_message_sent_failed reason=~1000p:~1000p",
+                         [Error, Reason]),
+            {FreshConn, FreshChann} = restart_rabbit_connection(Opts),
+            {noreply, Opts#state{connection = FreshConn, channel = FreshChann}};
+        timeout ->
+            update_messages_timeout_metrics(Host),
+            ?WARNING_MSG("event=rabbit_message_sent_failed reason=timeout",
+                         []),
+            {noreply, Opts}
+    end.
+
+-spec publish_message_and_wait_for_confirm(Method :: mongoose_amqp:method(),
+                                           Payload :: mongoose_amqp:message(),
+                                           worker_opts()) ->
+    boolean() | timeout | channel_exception.
+publish_message_and_wait_for_confirm(Method, Payload,
+                                     #state{channel = Channel,
+                                            confirms = IsConfirmEnabled}) ->
+    case try_to_apply(amqp_channel, call, [Channel, Method, Payload]) of
+        {ok, _} ->
+            maybe_wait_for_confirms(Channel, IsConfirmEnabled);
+        {error, E, R} -> {channel_exception, E, R}
+    end.
+
+-spec maybe_wait_for_confirms(Channel :: pid(), boolean()) ->
+    boolean() | timeout.
+maybe_wait_for_confirms(Channel, true) ->
+    amqp_channel:wait_for_confirms(Channel);
+maybe_wait_for_confirms(_, _) -> true.
+
+-spec restart_rabbit_connection(worker_opts()) -> {pid(), pid()}.
+restart_rabbit_connection(#state{connection = Conn, channel = Chann,
+                                 host = Host, amqp_client_opts = AMQPOpts}) ->
+    catch close_rabbit_connection(Conn, Chann, Host),
```

Comment:
The code ignores all the errors, the catch would ignore such errors as "Conn = undefined" for example.